Assignment4/NumberSet/NumSet.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace NumberSet
+{
+    public class NumSet
+    {
+        private HashSet<int>? numberSet;
+
+
+        public NumSet(params int[] ? numSet)
+        {
+            if(numSet is null)
+            {
+                throw new ArgumentNullException(nameof(NumSet));

```

Comment:
```suggestion
                throw new ArgumentNullException(nameof(numSet));
```
You should throw the name of what causes the null exception, in this case, it is the `numSet` parameter you are passing in.